### PR TITLE
[Bug Fix] LPS-107347

### PR DESCRIPTION
**Pull request by yuhai**
Hi @drewbrokke, 

This is an implementation from slack [talk](https://liferay.slack.com/archives/CNUCYF7EW/p1642778927010600?thread_ts=1641437137.000100&cid=CNUCYF7EW). Could you please help review this?

cc @dantewang

Thanks,
Hai

### CODE REVIEW

**Comment by epark18** [comment]
Hello, it looks there are a lot of failing checks due to an issue with password reset. Every test is failing at this step due to using the same password as both the original and the new one.  [Case Results](https://testray.liferay.com/home/-/testray/case_results/1604949398)  | [Error Logs](https://testray.liferay.com/reports/production/logs/2022-01/test-1-29/test-portal-acceptance-pullrequest(master)/1499/functional-tomcat90-mysql57-jdk8/0/19/jenkins-console.txt.gz) | [Poshi Report](https://testray.liferay.com/reports/production/logs/2022-01/test-1-29/test-portal-acceptance-pullrequest(master)/1499/functional-tomcat90-mysql57-jdk8/0/19/LocalFile.WorkflowMetrics_ViewReports/index.html.gz)
image

File: portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java [hunk 1]
```diff
@@ -318,7 +318,16 @@ public User addDefaultAdminUser(
 		updateLastLogin(
 			defaultAdminUser.getUserId(), defaultAdminUser.getLoginIP());
 
-		updatePasswordReset(defaultAdminUser.getUserId(), false);
+		boolean passwordReset = false;
+
+		PasswordPolicy passwordPolicy =
+			_passwordPolicyLocalService.getDefaultPasswordPolicy(companyId);
+
+		if ((passwordPolicy != null) && passwordPolicy.isChangeable()) {
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java:326]: I don't think that the password policy will ever be null. `_passwordPolicyLocalService.getDefaultPasswordPolicy` will throw an exception so you'd never reach this line in that case. I think the approach should be to surround the `get` call and `if` blocks with a `try/catch` statement which logs a warning if the password policy is not found. That way we don't introduce unexpected behavior.

Comment (drewbrokke) [reply portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java:326]: I think we should include `passwordPolicy.isChangeRequired()` in this `if` statement. This matches the logic found elsewhere in this class (see [UserLocalServiceImpl#addUserWithWorkflow](https://github.com/liferay/liferay-portal/blob/14504050c88c4fe140975994b613bd6fdefed7e5/portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java#L1238-L1247) for an example).

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@yuhai Nice work! I think a couple of adjustments are needed. Please see my inline comments.

Also, it looks like there may be some functional test failures. It's possible we may need to update the functional tests to account for this change, but would you be willing to take a look at those just in case? Thank you

**Comment by yuhai** [comment]
resend it to https://github.com/liferay-usm/liferay-portal/pull/442

**Pull request closed without merging**